### PR TITLE
fix(processor): show correctly progress in cluster mode

### LINKED ISSUE #6237: bug: incorrect progress summary in cluster mode
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

main

### What's Wrong?

```
mysql> SELECT number % 3, count() FROM numbers(1000000) GROUP BY number % 3;
+--------------+---------+
| (number % 3) | count() |
+--------------+---------+
|            1 |  333333 |
|            0 |  333334 |
|            2 |  333333 |
+--------------+---------+
3 rows in set (0.09 sec)
Read 0 rows, 0.00 B in 0.058 sec., 0 rows/sec., 0.00 B/sec.
```

### How to Reproduce?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix(processor): show correctly progress in cluster mode

## Changelog

- Bug Fix

## Related Issues

Fixes #6237

### CODE REVIEW

**Comment by BohuTANG** [comment]
Is this patch related to https://github.com/datafuselabs/databend/issues/5973?

**Comment by zhang2014** [comment]
> Is this patch also related to https://github.com/datafuselabs/databend/issues/5973?

unrelated.

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
I think this PR is not ready for review, so convert it to the draft.

File: query/src/api/rpc/exchange/exchange_channel_sender.rs [hunk 1]
```diff
@@ -12,66 +12,292 @@
...
+                    "Send result progress values error, because flight connection is closed."
+                );
+            }
+        }
+    }
+
+    async fn send_precommit_if_need(ctx: &Arc<QueryContext>, c_tx: &Sender<DataPacket>) {
+        let precommit_blocks = ctx.consume_precommit_blocks();
+
+        if !precommit_blocks.is_empty() {
```
Comment (sundy-li) [inline query/src/api/rpc/exchange/exchange_channel_sender.rs:203]: no need this judge I think,

Comment (zhang2014) [reply query/src/api/rpc/exchange/exchange_channel_sender.rs:203]: precommit_blocks is empty if execute `SELECT`;

**Review by sundy-li — APPROVED** [review]

**Pull request merged**